### DISCUSSION TITLE
libp2p-rPGF-1: Nominations for Other

### DISCUSSION THREAD

**Discussion by p-shahi** [libp2p/libp2p#214] [RetroPGF]
### Other Nominations

This discussion post is for projects nominations to the Other category.
This category is intended for projects that may not fit into other categorizes but are still a good candidate for libp2p-RPGF-1 funding.

### Nomination Template
When nominating a project, please answer the following questions in a comment to this discussion:

```
Project name:
The project name

Description:
A description of the positive impact of this project for this specific category over the past year (March 2023 - March 2024).

Github/Twitter link:
A link to the project’s GitHub or Twitter

Contact:
Contact info for the project or project lead
```

Things to consider when nominating a project:

- Describe how this project has delivered impact to the libp2p ecosystem in the previous year (March 2023 - March 2024). Use the provided template.
- Nominate projects in **one** relevant category. Do not nominate the same project across different categories! If a project has overlap in multiple categories, please choose just one category that best aligns with it. 
- Organizers of libp2p-RPGF-1 may reassign a project’s category from its nominated one if it better aligns elsewhere.
- Check if a project has been nominated already to avoid duplication.
- Please provide feedback to help the process improve in each round. Your input and feedback is valuable and you can use this post to submit feedback now.

Nominations will close on May 3rd, 2024.

See https://github.com/libp2p/libp2p/discussions/209 for an overview of the process, timelines, and to ask general questions.

**Comment by CookingCryptos** [comment]
Project name:
The DeFi Collective

Description:
Since its inception in October 2023, The DeFi Collective has rapidly become an instrumental entity within the Optimism ecosystem. As a non-profit organization, we are dedicated to enhancing and supporting DeFi protocols not only across multiple blockchains including Ethereum, Polygon, and Base, but with a focused impact on Optimism over the past year.
From March 2023 to March 2024, The DeFi Collective has actively contributed to the growth and stability of the Optimism network through targeted initiatives:

**Liquidity Enhancement:** We have strategically deployed liquidity solutions on Optimism. This includes significant support for the liquidity of POOL and HAI tokens on the Velodrome protocol. By utilizing our assets, voting power and bribes, we have helped minimize slippage and improve price stability, ensuring that these tokens remain viable and efficient for trading activities. In supporting these projects, we aid in maintaining a substantial and healthy liquidity pool, essential for their ongoing success and utility.

**Community Engagement and Education:** The DeFi Collective’s involvement extends beyond contributions in liquidity. Our active participation in governance, research, and community initiatives aligns with Optimism’s goals of fostering an engaged and innovative DeFi community.

**Governance and Collaboration:** Our active participation in governance has fostered robust collaborations among DeFi projects. We actively work to facilitate interactions that lead to mutual support and technological integration between projects we back. A recent highlight was PoolTogether's decision to allocate POOL tokens to The DeFi Collective, inspired by our proven ability to enhance liquidity. These tokens have already been leveraged to create new liquidity pools for POOL on multiple chains, significantly expanding its accessibility and trading efficiency. A notable contribution is our [proposal](https://commonwealth.im/lets-get-hai/discussion/16622-improve-hai-kite-liquidity-strategy-thanks-to-several-optimizations-fee-adjustment-for-kite-haikite-hailusd-pools-haieth-pool-migration) to refine the liquidity strategy for HAI & KITE, involving critical adjustments and migrations to optimize pool performance.

These efforts underscore our commitment to fostering a sustainable and thriving DeFi environment. Through our work, we aim to empower users and strengthen the foundational aspects of decentralization on Optimism and throughout the broader DeFi ecosystem..
Our efforts are dedicated to driving support and community growth through strategic initiatives that resonate with the core principles of both the DeFi Collective and the Optimism chain. As we advance into the next year, The DeFi Collective is eager to intensify its support for new projects and continue developing innovative solutions that enhance the functionality and accessibility of DeFi on Optimism.

Furthermore, these collaborative efforts reflect our broader mission to cultivate a healthier and more sustainable DeFi environment. Established from generous community donations, The DeFi Collective continues to thrive and expand its impact through ongoing financial support from the community. This model enables us to maintain our independence and commitment to fostering a decentralized financial ecosystem that upholds transparency and user-centric practices.


Github/Twitter link:
https://github.com/deficollective
https://twitter.com/DeFiCollective_



Contact:
https://twitter.com/CookingCryptos
https://twitter.com/TokenBrice

**Comment by haydenyoung** [comment]
Project name:
OrbitDB

Description:
OrbitDB is a peer-to-peer, decentralized, open source database system. OrbitDB's Sync protocol is built entirely on Libp2p, using Libp2p to ensure eventual consistency across databases on multiple peers.

Between March 2023 and March 2024, OrbitDB has managed a production release with integration of the latest Libp2p and an upgrade to Helia (also leveraging Libp2p). OrbitDB has received wide usage over the past 12 months and it could be considered one of the flagship projects of Libp2p.

Github/Twitter link:
https://github.com/orbitdb/orbitdb/

Contact:
Hayden Young @haydenyoung (info@orbitdb.org)

**Comment by tolbrino** [comment]
Project name:

HOPR

Description:

HOPR is a decentralized, incentivized, peer-to-peer mixnet which provides metadata-privacy for data flowing through the HOPR network. The inter-peer communication of HOPRd nodes is built on top of libp2p. Over the last year HOPRd migrated from being a NodeJS application to a native Rust application. That meant first migrating from `js-libp2p` to `rust-libp2p` via WASM as an intermediate step. The underlying TCP transport module was contributed to `rust-libp2p`, but not merged due to `wasm-ext` becoming deprecated.

https://github.com/libp2p/rust-libp2p/pull/4253

Nevertheless, the current stable version of HOPRd (2.0.8) is still running this setup with ca. 500 nodes live in the network. Therefore it is possibly a good reference for other developer on how to use `rust-libp2p` via WASM in production. The next version of HOPRd (2.1.0) will run natively using `rust-libp2p`.

As HOPR becomes a native application we are planning to move closer to recent `rust-libp2p` development and make further contributions.

Github/Twitter link:

https://github.com/hoprnet/hoprnet/
https://twitter.com/hoprnet

Contact:

https://github.com/tolbrino

e-mail at [libp2p@hoprnet.org](mailto:libp2p@hoprnet.org)

**Comment by marcus-pousette** [comment]
Project name:
Peerbit

Description:
Peerbit is p2p application framework built on top of libp2p

During period March 2023 - March 2024 development a lot of time has been spent into trying to make `js-libp2p` to be as fast as possible and develop protocols and their handlers on libp2p to enable users to build impactful applications

Some outcomes of this work is (that is beneficial for the libp2p ecosystem):

https://github.com/dao-xyz/js-libp2p-noise (A fork of chainsafe/js-libp2p-noise) that uses other crypto library integrations to improve stream encryption performance

Some benchmarking of the mplex and yamux libs. And many attempts of rewriting it in other ways of making it faster to match the `mplex` performance. But no significant success so far. 
https://github.com/ChainSafe/js-libp2p-yamux/issues/42

Benchmarking of protobufjs libraries like `protons`, `protobufjs` and development of an alternative encoding library [borsh-ts](https://github.com/dao-xyz/borsh-ts) which is an encoding that follows the `Borsh` specification and has a simple integration with Typescript that allows for faster development and better performing protocol handlers with `js-libp2p`. It also enables you to have properties with Enums that can have variants that can hold data (like how Enums behave in Rust)

Research and development of an alternative PubSub protocol (https://github.com/dao-xyz/peerbit/tree/master/packages/transport/pubsub) that supports some unique features
- Path optimization. Approximations of the shortest path to deliver packages through a network
- Redundancy options. Allow peers to specificy how many paths you want to attempt to deliver data packages on, i.e. allows for sending data packages in the `N` fastest path to subscribers to a specific topic
- Packet prioritizations. Allow some packages flow through the network faster than others to prevent critical functionality to stop in case of network congestion. 


Reporting (and trying to help to solve) issues related to `js-libp2p` (17)
https://github.com/libp2p/js-libp2p/issues?q=author%3Amarcus-pousette 

Adding support for sending arbitrarily large messages with `js-libp2p-webrtc` 
https://github.com/libp2p/js-libp2p-webrtc/pull/147

Onboarding and helping users work with `libp2p` with various configurations as they get exposed to it when using the `Peerbit` client

Github/Twitter link:
[Github](https://github.com/dao-xyz/peerbit)

Contact:
marcus@dao.xyz
